### ISSUE TITLE
Bug: Can't configure project service for files with custom extensions

### ISSUE THREAD

**Issue by tjzel** [eslint/eslint#20219]
### Environment

Node version: v22.14.0
npm version: v10.9.2
Local ESLint version: v9.37.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 24.6.0

### What parser are you using?

@typescript-eslint/parser

### What did you do?

I'm trying to configure ESLint in a repository which has platform-specific versions of files, i.e. there is:

- `Foo.tsx`
- `Foo.web.tsx`
- `Bar.tsx`
- `Bar.native.tsx`

However, I can't configure ESLint to properly lint them and I'm stuck on `Parsing error: /Users/user/eslint-repro/src/Foo.web.tsx was not found by the project service. Consider either including it in the tsconfig.json or including it in allowDefaultProject`.

The key aspect here is that I have several `tsconfig.json` files, respective to each platform.

<details><summary>Simplified TS configs look like this</summary>
<p>

```ts
// tsconfig.native.json
{
  "compilerOptions": {
    "moduleSuffixes": [".native", ".ios", ".android", ""] // disallow .web files
  },
  "include": ["src"],
  "exclude": ["src/**/*.web.ts", "src/**/*.web.tsx"]
}

// tsconfig.web.json
{
  "compilerOptions": {
    "moduleSuffixes": [".web", ""] // disallow .native, .ios, .android files
  },
  "include": ["src", "index.ts"],
  "exclude": [
    "src/**/*.native.ts",
    "src/**/*.native.tsx",
    "src/**/*.ios.ts",
    "src/**/*.ios.tsx",
    "src/**/*.android.ts",
    "src/**/*.android.tsx"
  ]
}

// tsconfig.json (default for IDE features)
{
  "extends": "./tsconfig.native.json"
}
```

</p>
</details> 

<details><summary>This is my base config that doesn't work</summary>
<p>

```ts
import tsEslint from "typescript-eslint";
import { defineConfig } from "eslint/config";
import importPlugin from "eslint-plugin-import";

/** @type {import('typescript-eslint').ConfigWithExtends[]} */
export default defineConfig(
  tsEslint.configs.recommendedTypeChecked,
  importPlugin.flatConfigs.recommended,
  {
    languageOptions: {
      parserOptions: {
        projectService: true,
      },
    },
    settings: {
      "import/resolver": {
        typescript: {
          alwaysTryTypes: true,
        },
      },
    },
  }
);
```

</p>
</details>

I've tried various options like explicit ignores, toying with `projectService.defaultProject` but nothing seems to work.

<details><summary>More fine grained config</summary>
<p>

```ts
import tsEslint from "typescript-eslint";
import { defineConfig } from "eslint/config";
import importPlugin from "eslint-plugin-import";

/** @type {import('typescript-eslint').ConfigWithExtends[]} */
export default defineConfig(
  {
    languageOptions: {
      parserOptions: {
        projectService: true,
      },
    },
    settings: {
      "import/resolver": {
        typescript: {
          alwaysTryTypes: true,
        },
      },
    },
    extends: [
      tsEslint.configs.recommendedTypeChecked,
      importPlugin.flatConfigs.recommended,
    ],
    files: ["**/*.ts", "**/*.tsx"],
    ignores: ["**/*.web.ts", "**/*.web.tsx"],
  },
  {
    languageOptions: {
      parserOptions: {
        projectService: {
          defaultProject: "tsconfig.web.json",
        },
      },
    },
    settings: {
      "import/resolver": {
        typescript: {
          alwaysTryTypes: true,
          project: "tsconfig.web.json",
        },
      },
    },
    extends: [
      tsEslint.configs.recommendedTypeChecked,
      importPlugin.flatConfigs.recommended,
    ],
    files: ["**/*.web.ts", "**/*.web.tsx"],
    ignores: ["**/*.native.ts", "**/*.native.tsx"],
  }
);
```

</p>
</details> 

### What did you expect to happen?

All files:

- `Foo.tsx`
- `Foo.web.tsx`
- `Bar.tsx`
- `Bar.native.tsx`

are linted when I invoke `yarn eslint src`

### What actually happened?

> Parsing error: /Users/user/eslint-repro/src/Foo.web.tsx was not found by the project service. Consider either including it in the tsconfig.json or including it in allowDefaultProject.

### Link to Minimal Reproducible Example

https://github.com/tjzel/eslint-parser-repro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by DMartens** [comment]
The error indicates that the `projectServices` only includes the `tsconfig.json` and not `tsconfig.web.json` and as such the files with `.web` suffix are excluded.
You could use the [project](https://typescript-eslint.io/packages/parser#project) parser option to explicitly list the `tsconfig.json` you want to include.
But the problem you are having is from the third-party [typescript-eslint](https://typescript-eslint.io/) and its parser services, so please use their support channels for further assistance.

**Comment by tjzel** [comment]
Thanks for the reply @DMartens. The `project` option doesn't work either.

I re-opened this issue in typescript-eslint gh as per your suggestion https://github.com/typescript-eslint/typescript-eslint/issues/11707
